File: app/src/main/java/kandroid/data/EquipmentData.java
package kandroid.data;

import kandroid.observer.POJO;
import kandroid.utils.Identifiable;

public class EquipmentData extends Data<EquipmentData.ApiSlotItem> implements Identifiable {

    public int getMasterID() {
        return data.api_id;
    }

    public int getEquipmentID() {
        return data.api_slotitem_id;
    }

    public boolean isLocked() {
        return data.api_locked != 0;
    }

    public int getLevel() {
        return data.api_level;
    }

    public int getAircraftLevel() {
        return data.api_alv;
    }

    public Start2Data.MasterEquipmentData getMasterEquipment() {
        return KCDatabase.INSTANCE.getMaster().masterEquipmentData.get(getMasterID());
    }

    public String getName() {
        return getMasterEquipment().getName();
    }

    public String getNameWithLevel() {
        throw new UnsupportedOperationException();
    }

    @Override
    public int getId() {
        return data.api_id;
    }

    public static class ApiSlotItem extends POJO {
        public int api_id;
        public int api_slotitem_id;
        public int api_locked;
        public int api_level;
        public int api_alv;
    }
}

File: app/src/main/java/kandroid/data/EquipmentData.kt
package kandroid.data

import com.google.gson.JsonElement
import kandroid.data.Start2Data.MasterEquipmentData
import kandroid.data.JsonWrapper
import kandroid.observer.kcsapi.api_req_kousyou
import kandroid.utils.CatException
import kandroid.utils.Identifiable
import kandroid.utils.json.get
import kandroid.utils.json.int
import kandroid.utils.json.set

class EquipmentData : JsonWrapper(), Identifiable {

    /**
     * 装备固有id
     */
    val masterID: Int get() = data["api_id"].int()
    /**
     * master数据Id
     */
    val equipmentID: Int get() = data["api_slotitem_id"].int()
    /**
     * 锁了?
     */
    val isLocked: Boolean get() = data["api_locked"].int() != 0
    /**
     * 改修星数
     */
    val level: Int get() = data["api_level"].int()
    /**
     * 老司机熟练度
     */
    val aircraftLevel: Int get() = data["api_alv"].int()
    /**
     * master数据
     */
    val masterEquipment: MasterEquipmentData get() = KCDatabase.master.masterEquipmentData[masterID]!!

    val name: String get() = masterEquipment.name

    val nameWithLevel: String get() = if (level > 0) {
        if (aircraftLevel > 0) "$name+$level Lv. $aircraftLevel" else "$name+$level"
    } else {
        if (aircraftLevel > 0) "$name Lv. $aircraftLevel" else name
    }

    override val id: Int get() = masterID

    override fun toString(): String = nameWithLevel

    override fun loadFromResponse(apiName: String, responseData: JsonElement) {
        // 由于GSON轮子的缘故, 只需要api_id和api_slotitem_id即可正常运作 TODO 复查
        super.loadFromResponse(apiName, responseData)
    }
}